### ISSUE TITLE
Recalculate feature importance during the update process of a tree model

### ISSUE THREAD

**Issue by Lean-Y** [microsoft/LightGBM#2413]
Feature importance ranking was calculated based on information about the loss gains learned within training data, which could carry significant overfitting and may unfairly inflate some importances. I want to estimate the local importance in the validation data using re-calculated gains.

Xgboost implements this feature, but I haven't found a similar feature in the document of lightgbm, "refit" can't recalculate the importance of features and will change the output of the leaf.

https://github.com/dmlc/xgboost/pull/1670

**Comment by StrikerRUS** [comment]
Closed in favor of being in #2302. We decided to keep all feature requests in one place.

Welcome to contribute this feature! Please re-open this issue (or post a comment if you are not a topic starter) if you are actively working on implementing this feature.
